fhir-model/src/test/java/com/ibm/fhir/model/test/ResourceCoverageTest.java

```diff
@@ -0,0 +1,260 @@
+                Class<?>[] clzs = resource.getClass().getClasses();
+                for (Class<?> clz : clzs) {
+
+                    if (!SKIP.contains(clz.getSimpleName())) {
+                        Method m = resource.getClass().getMethod("get" + clz.getSimpleName());
+                        Object o = m.invoke(resource);
+                        if (o.getClass().getSimpleName().contains("List")) {
+                            @SuppressWarnings("unchecked")
+                            List<Object> os = ((List<Object>) o);
+                            if (!os.isEmpty()) {
+                                runMethods(os.get(0), clz.getMethods());
+                            }
+                        } else if (!o.getClass().getSimpleName().contains("Class")) {
+                            runMethods(o, clz.getMethods());
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    @Test
+    public void testResourcesWithXml() throws Exception {
+        List<String> SKIP = Arrays.asList("DomainResource", "Resource", "Builder");
+        ValueSet[] values = ResourceType.ValueSet.values();
+        for (ValueSet valueSet : values) {
+            if (!SKIP.contains(valueSet.value())) {
+                Resource resource =
+                        TestUtil.readExampleResource("xml/ibm/complete-mock/" + valueSet.value() + "-1.xml");
+
+                Resource.Builder builder = resource.toBuilder();
+                Method[] methods = builder.getClass().getMethods();
+                for(Method method : methods) {
+                    if (method.getName().equals("extension") && method.toString().contains("java.util.Collection")) {
+                        List<Extension> extensions = Arrays.asList(buildBooleanTrue());
+                        builder = (Resource.Builder) method.invoke(builder, extensions);
+                    }
+                    
+                    if (method.getName().equals("modifierExtension") && method.toString().contains("java.util.Collection")) {
+                        List<Extension> extensions = Arrays.asList(buildBooleanTrue());
+                        builder = (Resource.Builder) method.invoke(builder, extensions);
+                    }
+                }
+                resource = builder.build();
+                
+                try (StringWriter writer = new StringWriter()) {
+                    FHIRGenerator.generator(Format.JSON).generate(resource, writer);
+                    String outJson = writer.toString();
+                    assertNotNull(outJson);
+                    assertFalse(outJson.isEmpty());
+                    
+                    try (ByteArrayInputStream in = new ByteArrayInputStream(outJson.getBytes())) {
+                        Resource resource2 = FHIRParser.parser(Format.JSON).parse(in).as(Resource.class);
+                        assertNotNull(resource2);
+                        assertTrue(true);
```

Comment:
always true?

Reply:
Yes.  that's right.  it's an invocation, and asserts true as a pattern.